inject-generator/src/main/java/io/avaje/inject/generator/TypeExtendsReader.java

```diff
@@ -227,7 +227,7 @@ private void readInterfacesOf(TypeMirror anInterface) {
           .noneMatch(g -> typeElement(g.mainType()) == null);
 
       interfaceTypes.add(knownType ? rawType : GenericType.removeParameter(rawType));
-      if (!rawType.startsWith("java.lang.")) {
+      if (!rawType.startsWith("java.lang.") || Character.isLowerCase(rawType.charAt(10))) {
```

Comment:
Not safe if rawType length is less than 10

Reply:
there's no situation where a java.lang class has less than 10 characters in the fqn